README.md

```diff
@@ -1,89 +1,167 @@
+is necessary (the default of `zram-fraction=0.5` will be used unless
+overriden), but the configuration file with at least one section must
+exist.
 
-### Mount points
+## Mount points
 
-```ini
-# /etc/systemd/zram-generator.conf
-[zram1]
-mount-point = /var/tmp
-```
+    # /etc/systemd/zram-generator.conf
+    [zram1]
+    mount-point = /var/tmp
 
-This will set up a /dev/zram1 with ext2 and generate a mount unit for /var/tmp.
+This will set up a /dev/zram1 with ext2 and generate a mount unit for
+/var/tmp.
 
-### Rust
+## Rust
 
 The second purpose of this program is to serve as an example of a
 systemd generator in rust. Details are still being figured out.
 
-### Installation
+## Installation
+
+### Package Manager
 
 It is recommended to use an existing package:
 
-* Fedora: `sudo dnf install zram-generator-defaults` (or `sudo dnf install zram-generator` to install without the default configuration)
-* Debian: packages provided by nabijaczleweli, see https://debian.nabijaczleweli.xyz/README.
-* Arch: `sudo pacman -S zram-generator` (or https://aur.archlinux.org/packages/zram-generator-git/ for the latest git commit)
+#### Fedora 33+
+
+    sudo yum install -y zram-generator
+
+#### Debian 11+ / Ubuntu 20.04+
+
+    curl -skL https://keybase.io/nabijaczleweli/pgp_keys.asc | sudo apt-key add
+    sudo apt-add-repository "deb https://debian.nabijaczleweli.xyz stable main"
+    sudo apt install -y systemd-zram
+
+#### Arch
+
+    sudo pacman -S zram-generator
+
+### Source
+
+#### Fedora 33+ / CentOS 8+ / RHEL 8+
+
+    sudo yum install -y cargo pkgconfig ruby-devel
+    sudo gem install ronn-ng
+
+#### Debian 10+ / Ubuntu 18.04+
+
+    sudo apt update
+    sudo apt install -y cargo pkg-config ruby-dev
+    sudo gem install ronn-ng
+
+#### openSUSE Leap 15.3+ / Tumbleweed
+
+    sudo zypper install -y cargo pkgconfig ruby-devel
+    sudo gem install ronn-ng
+    sudo ln -s /usr/bin/ronn.* /usr/bin/ronn
+
+#### Build and Install
+
+Get the latest source code with:
+
+    git clone https://github.com/systemd/zram-generator.git
+    cd zram-generator
+
+To install directly from sources, execute:
+
+    make
+    make test
+    sudo make install
+
+Which:
+
+  - `zram-generator` binary is installed in the systemd system generator
+    directory (usually `/usr/lib/systemd/system-generators/`)
+  - `zram-generator(8)` and `zram-generator.conf(5)` manpages are
+    installed into `/usr/share/man/manN/`, this requires
+    [`ronn`](https://github.com/apjanke/ronn-ng).
+  - `units/systemd-zram-setup@.service` is copied into the systemd
+    system unit directory (usually `/usr/lib/systemd/system/`)
+  - `zram-generator.conf.example` is copied into
+    `/usr/share/doc/zram-generator/` You need though create your own
+    config file at one of the locations listed above.
+
+Ensure required kernel module load during system start:
+
+    cat > /usr/lib/modules-load.d/zram-generator.conf <<-EOF
```

Comment:
Here and elsewhere, I'd prefer is redirections were not separated by a space, i.e.
`cat >/some/foo` rather than `cat > /some/foo`.

This part doesn't seem necessary. The generator creates `/run/modules-load.d/zram-generator.conf`.

Reply:
redo with #86